src/crc32.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2024 Space Cubics, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <csp/csp.h>
+#include <csp/csp_crc32.h>
+#include <csp/csp_id.h>
+#include <endian.h>
+
+struct crc_fixture {
+	csp_packet_t *buf;
+};
+
+static struct crc_fixture crc_fixture;
+
+static void *setup(void)
+{
+	csp_init();
+
+	return &crc_fixture;
+}
+
+static void before(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	fixture->buf = csp_buffer_get(0);
+	zassert_true(fixture->buf != NULL, NULL);
+}
+
+static void after(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	csp_buffer_free(fixture->buf);
+}
+
+ZTEST_F(crc, test_crc_verify_without_crc)
+{
+	csp_packet_t *packet = fixture->buf;
+	packet->length = sizeof(uint32_t)-1;
+	int ret;
+
+	/* check error if length < sizeof(crc) */
+	ret = csp_crc32_verify(packet);
+	zassert_equal(ret, CSP_ERR_CRC32, "Error on csp_crc32_verify");
+}
+
+ZTEST_F(crc, test_crc_without_header)
+{
+	csp_packet_t *packet = fixture->buf;
+	int ret;
+
+	memcpy(packet->data, "Hello", 5);
+	packet->length = 5;
+
+	/* Check without header */
+	csp_crc32_append(packet);
+	ret = csp_crc32_verify(packet);
+	zassert_equal(CSP_ERR_NONE, ret, "Error on csp_crc32_verify");
+}
+
+ZTEST_F(crc, test_crc_with_header)
+{
+	csp_packet_t *packet = fixture->buf;
+	uint32_t crc;
+	int ret;
+
+	memcpy(packet->data, "Hello", 5);
+	packet->length = 5;
+
+	/* Check with header.
+	* This simulate CSP_21 define on csp_crc32_append().
```

Comment:
Please format the code by zephry's clang-format.
Maybe you need to fix the below.

```
diff --git a/src/crc32.c b/src/crc32.c
index c206407..ec41a62 100644
--- a/src/crc32.c
+++ b/src/crc32.c
@@ -40,7 +40,7 @@ static void after(void *data)
 ZTEST_F(crc, test_crc_verify_without_crc)
 {
        csp_packet_t *packet = fixture->buf;
-       packet->length = sizeof(uint32_t)-1;
+       packet->length = sizeof(uint32_t) - 1;
        int ret;
 
        /* check error if length < sizeof(crc) */
@@ -72,8 +72,8 @@ ZTEST_F(crc, test_crc_with_header)
        packet->length = 5;
 
        /* Check with header.
-       * This simulate CSP_21 define on csp_crc32_append().
-       */
+        * This simulate CSP_21 define on csp_crc32_append().
+        */
        csp_id_prepend(packet);
        crc = csp_crc32_memory(packet->frame_begin, packet->frame_length);
        /* Convert to network byte order */
```